src/WebJobs.Script/Rpc/RpcInitializationService.cs

```diff
@@ -69,28 +84,52 @@ internal async Task InitializeRpcServerAsync()
 
         internal Task InitializeChannelsAsync()
         {
-            string workerRuntime = _environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName);
-            if (_environment.IsLinuxAppServiceEnvironment())
+            if (ShouldStartInPlaceholderMode())
             {
-                return Task.CompletedTask;
+                return InitializePlaceholderChannelsAsync();
             }
-            if (_environment.IsLinuxContainerEnvironment())
+            else
             {
-                return Task.CompletedTask;
+                return InitializeWebHostRuntimeChannelsAsync();
             }
-            if (string.IsNullOrEmpty(workerRuntime) && _environment.IsPlaceholderModeEnabled())
+        }
+
+        private Task InitializePlaceholderChannelsAsync()
+        {
+            if (_environment.IsLinuxHostingEnvironment())
             {
-                // Only warm up language workers in placeholder mode in worker runtime is not set
-                return Task.WhenAll(_languages.Select(runtime => _languageWorkerChannelManager.InitializeChannelAsync(runtime)));
+                InitializePlaceholderChannelsAsync(OSPlatform.Linux);
             }
-            if (_languages.Contains(workerRuntime))
+            else
             {
-                return _languageWorkerChannelManager.InitializeChannelAsync(workerRuntime);
+                InitializePlaceholderChannelsAsync(OSPlatform.Windows);
             }
+
             return Task.CompletedTask;
         }
 
+        private Task InitializePlaceholderChannelsAsync(OSPlatform os)
+        {
+            return Task.WhenAll(_hostingOSToWhitelistedRuntimes[os].Select(runtime =>
+                _languageWorkerChannelManager.InitializeChannelAsync(runtime)));
+        }
+
+        private Task InitializeWebHostRuntimeChannelsAsync()
+        {
+            if (_webHostLevelWhitelistedRuntimes.Contains(_workerRuntime))
+            {
+                return _languageWorkerChannelManager.InitializeChannelAsync(_workerRuntime);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private bool ShouldStartInPlaceholderMode()
```

Comment:
Nit: since there is only one reference to this method, remove method and check condition in if statement directly

Reply:
I'd like to keep this as it is because it makes it clear what the intention of the conditional check is. The alternative is to add a comment with the same information but I prefer to document via code vs. comments (people forget to change comments when they change code - speaking as someone who has forgotten to do it before 😉)